scripts/ci-conformance.sh

```diff
@@ -0,0 +1,191 @@
+#!/bin/bash
+
+# Copyright 2020 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+###############################################################################
+
+# To run locally, set AZURE_CLIENT_ID, AZURE_CLIENT_SECRET, AZURE_SUBSCRIPTION_ID, AZURE_TENANT_ID
+
+set -o nounset
+set -o pipefail
+
+REPO_ROOT=$(dirname "${BASH_SOURCE[0]}")/..
+cd "${REPO_ROOT}" || exit 1
+
+# shellcheck source=../hack/ensure-go.sh
+source "${REPO_ROOT}/hack/ensure-go.sh"
+# shellcheck source=../hack/ensure-kind.sh
+source "${REPO_ROOT}/hack/ensure-kind.sh"
+# shellcheck source=../hack/ensure-kubectl.sh
+source "${REPO_ROOT}/hack/ensure-kubectl.sh"
+# shellcheck source=../hack/ensure-kustomize.sh
+source "${REPO_ROOT}/hack/ensure-kustomize.sh"  
+
+random-string() {
+    cat /dev/urandom | tr -dc 'a-zA-Z0-9' | fold -w ${1:-32} | head -n 1
+}
+
+# generate manifests needed for creating the Azure cluster to run the tests
+add_kustomize_patch() {
+    # Enable the bits to inject a script that can pull newer versions of kubernetes
+    if ! grep -i -wq "patchesStrategicMerge" "templates/kustomization.yaml"; then
+        echo "patchesStrategicMerge:" >> "templates/kustomization.yaml"
+    fi
+    if ! grep -i -wq "kustomizeversions" "templates/kustomization.yaml"; then
+        echo "- kustomizeversions.yaml" >> "templates/kustomization.yaml"
+    fi
+}
+
+# build Kubernetes E2E binaries
+build_k8s() {
+    # possibly enable bazel build caching before building kubernetes
+    if [[ "${BAZEL_REMOTE_CACHE_ENABLED:-false}" == "true" ]]; then
+    create_bazel_cache_rcs.sh || true
+    fi
+
+    pushd "$(go env GOPATH)/src/k8s.io/kubernetes"
+
+    # make sure we have e2e requirements
+    bazel build //cmd/kubectl //test/e2e:e2e.test //vendor/github.com/onsi/ginkgo/ginkgo
+
+    # ensure the e2e script will find our binaries ...
+    mkdir -p "${PWD}/_output/bin/"
+    cp -f "${PWD}/bazel-bin/test/e2e/e2e.test" "${PWD}/_output/bin/e2e.test"
+    cp -f "${PWD}/bazel-bin/vendor/github.com/onsi/ginkgo/ginkgo/darwin_amd64_stripped/ginkgo" "${PWD}/_output/bin/ginkgo"
+    export KUBECTL_PATH="$(dirname "$(find "${PWD}/bazel-bin/" -name kubectl -type f)")/kubectl"
+    PATH="${KUBECTL_PATH}:${PATH}"
+    export PATH
+
+    # attempt to release some memory after building
+    sync || true
+    echo 1 > /proc/sys/vm/drop_caches || true
+
+    popd
+}
+
+create_cluster() {
```

Comment:
Is create cluster the same set of steps as the `manual-testing.sh`?  could the set of scripts be combined in one location?

Reply:
good point, it's pretty similar

Reply:
done